app/controllers/api_controller.rb

```diff
@@ -0,0 +1,92 @@
+class ApiController < ActionController::API
+
+  before_action :authenticate_member!
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+
+  protected
+
+  def authenticate_member!
+    return true if current_member
+    if authorization_token.present?
+      render api_response(
+               data: {}, success: false, message: 'Access is Forbidden', serializer: false
+             ).merge(status: :forbidden)
+    else
+      render api_response(
+               data: {}, success: false, message: 'Access Denied', serializer: false
+             ).merge(status: :unauthorized)
+    end
+  end
+
+  def render_invalid_response(message: 'invalid!', status: :not_acceptable)
+    render api_response(
+               serializer: false,
+               message: message,
+               data: {},
+               success: false
+             ).merge(status: status)
+  end
+
+  def authorization_token
+    request.headers['Authorization']
+  end
+
+  def current_member
+    @current_member ||= (
+      authorization_token.present? ? AccessToken.member_for(jwt: authorization_token) : nil
+    )
+  end
+
+  def api_response(data:,
+                   success: response.status < 300,
```

Comment:
Where is `response.status` set?

In the users/add endpoint there's this line:

`render api_response data: {}, serializer: false, message: "Failed to create user", success: false`

but I don't see status being set before this call.  `success` will be false but I don't know what the status code will be.

Reply:
Yes, I'm now thinking it doesn't make sense to set success from the response.status since that'll aways be 200 by default.  Plus api_response wasn't setting the response headers and it felt like it should. I've reworked it a bit to be more clear